pkg/common/peertracker/peertracker_test.go

```diff
@@ -70,17 +73,41 @@ func (p *PeerTrackerTestSuite) TestTrackerClose() {
 }
 
 func (p *PeerTrackerTestSuite) TestUDSListener() {
-	doneCh := make(chan struct{})
 	peer := newFakeUDSPeer(p.T())
 
-	peer.connect(p.unixAddr, doneCh)
+	sErrCh := make(chan error, 1)
+	connCh := make(chan net.Conn)
```

Comment:
If Accept() succeeds but then the timeout happens in between then and the write into `connCh` then the send `connCh` will be blocked.  If the channel is buffered (size of 1) then the send will succeed and the goroutine won't be leaked.